boot/bootutil/src/loader.c

```diff
@@ -659,31 +750,46 @@ boot_write_status(struct boot_status *bs)
  * Validate image hash/signature in a slot.
  */
 static int
-boot_image_check(struct image_header *hdr, const struct flash_area *fap,
-        struct boot_status *bs)
+boot_image_check(struct boot_loader_state *state, struct image_header *hdr,
+                 const struct flash_area *fap, struct boot_status *bs)
 {
-    static uint8_t tmpbuf[BOOT_TMPBUF_SZ];
+#if !defined(__BOOTSIM__)
```

Comment:
Perhaps have a single ifdef on `__BOOTSIM__` and define something like `SIMSTATIC` (or maybe `TARGET_STATIC`) as either `static` or empty, and then just use that symbol in the places where there is an ifdef, especially since there are a bunch of these.

Reply:
I have gone with `TARGET_STATIC`